Libraries/Components/ProgressViewIOS/ProgressViewIOS.ios.js

```diff
@@ -29,70 +25,59 @@ const RCTProgressView = requireNativeComponent('RCTProgressView');
 
 type Props = $ReadOnly<{|
   ...ViewProps,
+
+  /**
+   * The progress bar style.
+   */
   progressViewStyle?: ?('default' | 'bar'),
+
+  /**
+   * The progress value (between 0 and 1).
+   */
   progress?: ?number,
+
+  /**
+   * The tint color of the progress bar itself.
+   */
   progressTintColor?: ?ColorValue,
-  trackTintColor?: ?string,
+
+  /**
+   * The tint color of the progress bar track.
+   */
+  trackTintColor?: ?ColorValue,
+
+  /**
+   * A stretchable image to display as the progress bar.
+   */
   progressImage?: ?ImageSource,
+
+  /**
+   * A stretchable image to display behind the progress bar.
+   */
   trackImage?: ?ImageSource,
 |}>;
 
 /**
  * Use `ProgressViewIOS` to render a UIProgressView on iOS.
  */
-const ProgressViewIOS = createReactClass({
-  displayName: 'ProgressViewIOS',
-  mixins: [NativeMethodsMixin],
-
-  propTypes: {
-    ...DeprecatedViewPropTypes,
-    /**
-     * The progress bar style.
-     */
-    progressViewStyle: PropTypes.oneOf(['default', 'bar']),
-
-    /**
-     * The progress value (between 0 and 1).
-     */
-    progress: PropTypes.number,
-
-    /**
-     * The tint color of the progress bar itself.
-     */
-    progressTintColor: PropTypes.string,
-
-    /**
-     * The tint color of the progress bar track.
-     */
-    trackTintColor: PropTypes.string,
-
-    /**
-     * A stretchable image to display as the progress bar.
-     */
-    progressImage: Image.propTypes.source,
-
-    /**
-     * A stretchable image to display behind the progress bar.
-     */
-    trackImage: Image.propTypes.source,
-  },
-
-  render: function() {
-    return (
-      <RCTProgressView
-        {...this.props}
-        style={[styles.progressView, this.props.style]}
-      />
-    );
-  },
-});
+const ProgressViewIOS = (
+  props: Props,
+  forwardedRef?: ?React.Ref<'RCTProgressView'>,
```

Comment:
I don't know if this is right.

Reply:
This is the way the other components (like `Text`) seem to do it - but I notice that `View` does it a better way, so I updated it to do it the way `View` does it. The other ones should be cleaned up too, perhaps!